### PR TITLE
Swap cpu and memory in HPA template

### PR DESCRIPTION
**Pull request by coolbry95**
### Proposed changes
There is a kubernetes issue where the cpu and memory get swapped at some point. This make Argocd think the manifest is always out of sync. Swap them here so that Argocd is happy.

#3769

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: deployments/helm-chart/templates/controller-deployment.yaml [hunk 1]
```diff
@@ -10,7 +10,9 @@ metadata:
   annotations: {{ toYaml .Values.controller.annotations | nindent 4 }}
 {{- end }}
 spec:
+  {{- if not .Values.controller.autoscaling.enabled }}
```
Comment (lucacome) [inline deployments/helm-chart/templates/controller-deployment.yaml:13]: This is `1` by default, also not sure it should be connected to autoscaling

File: deployments/helm-chart/templates/controller-hpa.yaml [hunk 2]
```diff
@@ -18,20 +18,20 @@ spec:
   minReplicas: {{ .Values.controller.autoscaling.minReplicas }}
   maxReplicas: {{ .Values.controller.autoscaling.maxReplicas }}
   metrics:
-    {{- if .Values.controller.autoscaling.targetCPUUtilizationPercentage }}
+    {{- if .Values.controller.autoscaling.targetMemoryUtilizationPercentage }}
```
Comment (lucacome) [inline deployments/helm-chart/templates/controller-hpa.yaml:21]: I'm not really sure what difference it makes to swap the two. Do you have a link to any docs about this?

Comment (coolbry95) [reply deployments/helm-chart/templates/controller-hpa.yaml:21]: You can see the k8s issue here. https://github.com/kubernetes/kubernetes/issues/74099

The problem is somewhere after the manifest is applied in k8s the order swaps. This causes Argocd to think the manifest is always out of sync. By swapping these Argocd will not think the manifest is out of sync.

Comment (lucacome) [reply deployments/helm-chart/templates/controller-hpa.yaml:21]: Yeah I saw that, still seems pretty weird to me 🤷

**Review by lucacome — APPROVED** [review]

**Comment by brianehlert** [comment]
This fix follows prior art from here: https://github.com/kubernetes/kubernetes/issues/74099
Further discussed here: https://github.com/argoproj/argo-cd/issues/9145#issuecomment-1507820227
This fix is probably the lowest impact and has the effect of making ArgoCD content.

**Review by shaun-nx — APPROVED** [review]

**Pull request merged**